core/src/execution/datafusion/planner.rs

```diff
@@ -1026,6 +1106,16 @@ impl From<ExpressionError> for DataFusionError {
     }
 }
 
+/// Returns true if given operator can return input array as output array without
+/// modification. This is used to determine if we need to copy the input batch to avoid
+/// data corruption from reusing the input batch.
+fn op_reuse_array(op: &Arc<dyn ExecutionPlan>) -> bool {
```

Comment:
nit: I wonder if we can call this `can_reuse_input_batch`

Reply:
Ok